### PR TITLE
feat: introduce CSS filter API on FilterImage

### PR DESCRIPTION
**Pull request by jakex7**
# Summary

Provide a **CSS**-like filter API in the `FilterImage` component. 
Unlike the SVG filter API, which can be complex and challenging even for simple tasks, the CSS API is straightforward and allows users to quickly achieve satisfactory results.

The full API can be viewed here https://developer.mozilla.org/en-US/docs/Web/CSS/filter
_We support all `<filter-function>` listed in the docs while we do not support functions from `<url>`  (`url()` and `src()`)._

All shorthands are implemented according to the W3 standard described here
https://www.w3.org/TR/filter-effects-1/#ShorthandEquivalents

This PR also changes the `filters` prop behavior as it adds `fe` in front of `name` attribute to not introduce any abstract above that specified in the docs.
```tsx
<FilterImage
  source={myImage}
  filters={[{ name: 'colorMatrix', type: 'saturate', values: 0.2 }])
/>
```
is now 
```tsx
<FilterImage
  source={myImage}
  filters={[{ name: 'feColorMatrix', type: 'saturate', values: 0.2 }])
/>
```

## Examples
Below are the simple examples of the usage

through `StyleSheet`
```tsx
import React from 'react';
import {StyleSheet, View} from 'react-native';
import {FilterImage} from 'react-native-svg/filter-image';

export default () => {
  return (
    <FilterImage
      style={styles.image}
      source={{
        uri: 'https://cdn.pixabay.com/photo/2024/05/26/00/40/lizard-8787888_1280.jpg',
      }}
    />
  );
};
const styles = StyleSheet.create({
  image: {
    width: 200,
    height: 200,
    filter: 'saturate(100) grayscale(100%)',
  },
});
```

or directly in the `style` prop

```tsx
import React from 'react';
import {StyleSheet, View} from 'react-native';
import {FilterImage} from 'react-native-svg/filter-image';

export default () => {
  return (
    <FilterImage
      style={{
        width: 200,
        height: 200,
        filter: 'saturate(100) grayscale(100%)',
      }}
      source={{
        uri: 'https://cdn.pixabay.com/photo/2024/05/26/00/40/lizard-8787888_1280.jpg',
      }}
    />
  );
};
```

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added documentation in `README.md`
- [x] I updated the typed files (typescript)

### CODE REVIEW

File: src/filter-image/extract/extractFilters.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    return rawFilters as Filters;
+  }
+  return parse(rawFilters);
+};
+
+export const mapFilterToComponent = (
+  { name, ...props }: FilterElement,
+  index: number
+) => {
+  return tags[name]
```
Comment (WoLewicki) [inline src/filter-image/extract/extractFilters.ts:20]: We update the `tags` with the filters that are implemented yeah?

**Review by WoLewicki — COMMENTED** [review]
Left some comments, please answer them

File: package.json [hunk 2]
```diff
@@ -53,7 +53,7 @@
     "format-js": "prettier --write README.md CONTRIBUTING.md CODE_OF_CONDUCT.md USAGE.md ./src/**/*.{ts,tsx} ./apps/**/*.tsx ./example/**/*.tsx ./*-example/**/*.tsx",
     "jest": "jest",
     "lint": "eslint --ext .ts,.tsx src",
-    "peg": "pegjs -o src/lib/extract/transform.js ./src/lib/extract/transform.peg",
+    "peg": "pegjs -o src/lib/extract/transform.js ./src/lib/extract/transform.peg && peggy -o src/filter-image/extract/extractFiltersString.js src/filter-image/extract/extractFiltersString.pegjs",
```
[the remark this answers is not in the document]
Comment (jakex7) [reply package.json:56]: > Peggy is the successor of [PEG.js](https://github.com/pegjs/pegjs).
https://github.com/peggyjs/peggy?tab=readme-ov-file#peggy

I didn't want to change `pegjs` as the current `transform` implementation relies on it, however it should be compatible, and we can remove `pegjs` in the near future as a follow-up

File: src/filter-image/extract/extractFilters.ts (hunk 1 above)
Comment (jakex7) [reply src/filter-image/extract/extractFilters.ts:20]: Yes, `tags` are imported from `xml.ts` file, there we have a list of all implemented `SVG` components
```tsx
export const tags: { [tag: string]: ComponentType } = {
  svg: Svg,
  circle: Circle,
  ...,
  filter: Filter,
  feColorMatrix: FeColorMatrix,
};
```

File: apps/examples/src/examples/FilterImage/RemoteImage.tsx [hunk 3]
```diff
@@ -9,11 +9,7 @@ const testSource = {
 const FilterImageRemoteExample = () => {
   return (
     <View>
-      <FilterImage
-        filters={[{name: 'colorMatrix', type: 'saturate', values: [3]}]}
-        source={testSource}
-        style={{width: 200, height: 200}}
-      />
+      <FilterImage source={testSource} style={styles.image} />
```
[the remark this answers is not in the document]
Comment (jakex7) [reply apps/examples/src/examples/FilterImage/RemoteImage.tsx]: Added in [5b93a13](https://github.com/software-mansion/react-native-svg/pull/2359/commits/5b93a137c85ac9fcce72612497a3cd1984d4b087)

File: apps/examples/src/examples/FilterImage/LocalImage.tsx [hunk 4]
```diff
@@ -7,21 +7,18 @@ const testImage = require('../../assets/image.jpg');
 const FilterImageLocalExample = () => {
   return (
     <View>
-      <FilterImage
-        filters={[{name: 'colorMatrix', type: 'saturate', values: [0.5]}]}
-        source={testImage}
-      />
+      <FilterImage source={testImage} style={{filter: 'saturate(0.5)'}} />
```
[the remark this answers is not in the document]
Comment (jakex7) [reply apps/examples/src/examples/FilterImage/LocalImage.tsx:10]: Added in [5b93a13](https://github.com/software-mansion/react-native-svg/pull/2359/commits/5b93a137c85ac9fcce72612497a3cd1984d4b087)

**Review by WoLewicki — APPROVED** [review]
LGTM!

**Pull request merged**